### PR TITLE
Refactor collections to use show more + align with notifications tab.

### PR DESCRIPTION
**Pull request by Guyzeroth**
## Summary

- Refactor 'Bookmarks' into 'Collections'
- Align collections with notifications + watched tabs model.
- Add limit + offset paging to collections (Was page number paging)

Note
- This PR introduces a fair amount of code duplication which will be cleaned up in coming PRs.
- Corresponding Kuma changes can be found [https://github.com/mdn/kuma/pull/8048](url)

### CODE REVIEW

File: client/src/document/hooks.ts [hunk 1]
```diff
@@ -126,7 +126,7 @@ const sortByVisitsThenTimestampDesc = (
 };
 
 export function useFrequentlyViewed(): [
-  FrequentlyViewedEntry[],
+  any,
```
[the remark this answers is not in the document]
Comment (caugner) [reply client/src/document/hooks.ts:129]: Also below:

```
    const entries = JSON.parse(
      localStorage.getItem(FREQUENTLY_VIEWED_STORAGE_KEY) || "[]"
   );
```

can be changed to:

```
    const entries = JSON.parse(
      localStorage.getItem(FREQUENTLY_VIEWED_STORAGE_KEY) || "[]"
    ) as FrequentlyViewedEntry[];
```

Although we might need to be very careful here when changing the model.

File: client/src/plus/common/index.tsx [hunk 2]
```diff
@@ -31,3 +32,22 @@ export function DataError({ error }: { error: Error }) {
     </NoteCard>
   );
 }
+
+export function _getIconLabel(url) {
+  let category = docCategory({ pathname: url });
+
+  if (category) {
```
Comment (caugner) [inline client/src/plus/common/index.tsx:39]: Matter of taste, but I prefer to return early:

```suggestion
  if (!category) {
    return "docs":
  }
```

Makes the function easier to skim.
```

File: client/src/plus/common/index.tsx [hunk 3]
```diff
@@ -31,3 +32,22 @@ export function DataError({ error }: { error: Error }) {
     </NoteCard>
   );
 }
+
+export function _getIconLabel(url) {
+  let category = docCategory({ pathname: url });
```
Comment (caugner) [inline client/src/plus/common/index.tsx:37]: Here we could reuse the `getCategoryFromUrl()` function, suggested above.

File: client/src/plus/notifications/tabs.tsx [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
   WATCHING,
+  COLLECTIONS,
+  FREQUENTLY_VIEWED,
 }
 
 const ALL_URL = "/plus/notifications";
 const STARRED_URL = "/plus/notifications/starred";
 const WATCHING_URL = "/plus/notifications/watching";
+const COLLECTIONS_URL = "/plus/collection";
+const FREQUENTLY_VIEWEED_URL = "/plus/collection/frequently_viewed";
```
Comment (caugner) [inline client/src/plus/notifications/tabs.tsx:17]: Should these be in a more central `constants.tsx`, possibly properties of a `PlusRoutes` or `PlusPaths` object?

**Review by caugner — COMMENTED** [review]
Skimmed through the code, looking more closely here and there.

File: client/src/plus/collections/collection-list-item.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  onEditSubmit,
+  handleDelete,
+}: {
+  item: BookmarkData;
+  showEditButton: boolean;
+  onEditSubmit: CallableFunction;
+  handleDelete: (item: BookmarkData) => Promise<void>;
+}) {
+  const [show, setShow] = React.useState(false);
+  const [doomed, setDoomed] = React.useState(false);
```
Comment (fiji-flo) [inline client/src/plus/collections/collection-list-item.tsx:22]: reminder: cleanup

File: client/src/plus/common/index.tsx (hunk 3 above)
Comment (fiji-flo) [reply client/src/plus/common/index.tsx:37]: Let's do this later. https://github.com/mdn/yari/issues/5637

**Review by fiji-flo — APPROVED** [review]
Looks fine overall!

File: client/src/plus/collections/collection-list-item.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+  let className = "icon-card";
+  if (doomed) {
+    className += " doomed";
+  }
```
[the remark this answers is not in the document]
Comment (Guyzeroth) [reply client/src/plus/collections/collection-list-item.tsx:27]: Unfortunately not. 

`className?: string | undefined;`

File: client/src/plus/notifications/tabs.tsx (hunk 4 above)
Comment (Guyzeroth) [reply client/src/plus/notifications/tabs.tsx]: I have refactored this to make the tabs.tsx common for both consumers. I think in that context these constants can live there.

**Pull request merged**